IDEM-147 (#147)
dltxio/idem-mobile: README.md

### 2: New private key

> _Idem_ can automatically create a 256-bit private key on the device the user to import a mnemonic seed phrase (based on the bitcoin BIP39 standard) of their choice. This will be used to sign and verify requests (using elliptic curve crytography - Secp256k1) to third parties.

Reviewer: this doesn't make sense "private key on the device the user to import a mnemonic seed phrase"

Author: Fixed up in commit

> _Idem_ can automatically create a 256-bit private key on the device the user to import a mnemonic seed phrase (based on the bitcoin BIP39 standard) of their choice. This will be used to sign and verify requests (using elliptic curve crytography - Secp256k1) to third parties.

Author: Fixed up in commit

The user initiates the registration process by entering their email address and a password into the website. The use of Recaptchas on most websites rules out a full-fledged _Idem_ registration, although _Idem_ could be used to fill the email address and generate a password on behalf of the user. From here, the user can use _Idem_ to complete the verification process by allowing _Idem_ to supply the claims the website requires.

### 2. Existing (to the website) non-verified user

An existing user can log into the website, given the user is using the same email address as their claim in IDEM, the user will receive an alert stating "Would you like to use IDEM to verify with the website?" If the user accepts this alert they will be verified on the platform once their claims have been verified.

## Verification Workflow Diagram

The flowchart below is a verification workflow diagram for third-party developers to integrate a website (such as an exchange) with _Idem_. It works as follows:

1. A user creates account on the website, typically with a email and password.

> 2. The new user may then be assed to supply more information to meet KYC obligations. This could be meta data such as Full Name, Date of Birth and Phyiscal address. These are what Idem referrers to as _Claims_. To capture this data, the website can either

Reviewer: ```suggestion
2. The new user may then be assed to supply more information to meet KYC obligations. This could be meta data such as Full Name, Date of Birth and Phyiscal address. These are what Idem referrers to as _Claims_. To capture this data, the website can either.
```

> 2. The new user may then be assed to supply more information to meet KYC obligations. This could be meta data such as Full Name, Date of Birth and Phyiscal address. These are what IDEM referrers to as _Claims_. To capture this data, the website can either.

Reviewer 2: "assed"? haha "asked"? maybe

  - Get the user to fill out a form.
  - Present a self documenting QR code requesting the claims required, and the callback URL for the website. This is similar to a PayPal IPN or an OAuth2 callback URL.

> 5. The user is asked to confirm they're happy to proceed with sharing those specific claims to the website.
> 6. IDEM then posts the claims in DID format to the websites specified callback URL, which it can then add to the users profile.

Reviewer: ```suggestion
3. The user is asked to confirm they're happy to proceed with sharing those specific claims to the website.
4. IDEM then posts the claims in DID format to the websites specified callback URL, which it can then add to the users profile.
```

| **0x03** | Bank statement               | A bank statement issued in your name. Must be issued within the last 3 months.                                                                                       | Given name, middle name, family name, residential address                              | Secondary         | No                |
| **0x04** | Rates notice                 | A paid rates notice issued in your name with your address that is less than 12 months old.                                                                           | Given name, middle name, family name, residential address                              | Secondary         | No                |
| **0x05** | Utility account              | Water, gas, electricity or phone account with a receipt number. This must be in your name, show your address and be less than 12 months old. Must be a paid account. | Given name, middle name, family name, residential address                              | Secondary         | No                |

> | **0x06** | Medicare Card                | A current Medicare Card issued in your name.                                                                                                                         | Given name, middle name, family name, medicare number, expiry date, position in family | Secondary         | No                | 

Reviewer 3: If you're including the medicare card's expiry date, then should you also include expiry date for drivers licence and passport?

Author: Yes good point, not needed right now in the idem app. Will remove.

| CredentialSubject   | Mnemonic     | Standard            | Description            |
| ------------------- | ------------ | ------------------- | ---------------------- |
| AdultCredential     | eighteenplus | 18 Plus                             | 18 Plus                |
| BirthCredential     | dob          | YYYY-MM-DD ISO 8601                 | Users date of birth    |

> | NameCredential      | fullname     | given,name, family name, middle name| Users full name        |

Reviewer 2: Not sure about this little bit here "given,name, family name, middle name" does not need the "," between "given,name"

| EmailCredential     | email        | email@email.email                   | Users email address    |
| MobileCredential    | mobilenumber | Mobile Number                       | Users mobile number    |
| AddressCredential   | address      | Physical Address                    | Users physical address |

> | TaxCredential       | taxnumber    | 9 digit number                      | Users tax file number  |

Reviewer: ```suggestion
| TaxCredential       | taxnumber    | [0-9]{9}                      | Users tax file number  |
```
